crates/libcontainer/src/process/memory_policy.rs

```diff
@@ -0,0 +1,542 @@
+                .map_err(|_| MemoryPolicyError::InvalidNodes(format!("Invalid node: {}", range)))?;
+
+            node_ids.push(node);
+        }
+    }
+
+    Ok(node_ids)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::syscall::syscall::create_syscall;
+    use crate::syscall::test::TestHelperSyscall;
+
+    #[test]
+    fn test_parse_node_string() {
+        // Test empty string
+        assert_eq!(parse_node_string("").unwrap(), Vec::<u32>::new());
+
+        // Test single node
+        assert_eq!(parse_node_string("0").unwrap(), vec![0]);
+        assert_eq!(parse_node_string("1").unwrap(), vec![1]);
+        assert_eq!(parse_node_string("2").unwrap(), vec![2]);
+
+        // Test node range
+        assert_eq!(parse_node_string("0-2").unwrap(), vec![0, 1, 2]);
+        assert_eq!(parse_node_string("1-3").unwrap(), vec![1, 2, 3]);
+
+        // Test multiple nodes
+        assert_eq!(parse_node_string("0,2").unwrap(), vec![0, 2]);
+        assert_eq!(parse_node_string("0,1,3").unwrap(), vec![0, 1, 3]);
+
+        // Test combination of ranges and single nodes
+        assert_eq!(parse_node_string("0-1,3").unwrap(), vec![0, 1, 3]);
+        assert_eq!(parse_node_string("0,2-3").unwrap(), vec![0, 2, 3]);
+
+        // Test with spaces
+        assert_eq!(parse_node_string(" 0 , 2 ").unwrap(), vec![0, 2]);
+        assert_eq!(parse_node_string(" 0 - 2 ").unwrap(), vec![0, 1, 2]);
+
+        // Test whitespace-only string
+        assert_eq!(parse_node_string("   ").unwrap(), Vec::<u32>::new());
+        assert_eq!(parse_node_string(" , , ").unwrap(), Vec::<u32>::new());
+
+        // Test error cases
+        assert!(parse_node_string("2-1").is_err()); // Invalid range
+        assert!(parse_node_string("abc").is_err()); // Invalid format
+        assert!(parse_node_string("0-abc").is_err()); // Invalid range end
+    }
+
+    #[test]
+    fn test_setup_memory_policy() {
+        use oci_spec::runtime::{LinuxMemoryPolicyBuilder, MemoryPolicyModeType};
+
+        let syscall = create_syscall();
+
+        // Test with None (no memory policy)
+        assert!(setup_memory_policy(&None, syscall.as_ref()).is_ok());
+
+        // Test with basic memory policy
+        let policy = LinuxMemoryPolicyBuilder::default()
+            .mode(MemoryPolicyModeType::MpolBind)
+            .nodes("0,1".to_string())
+            .build()
+            .unwrap();
+
+        assert!(setup_memory_policy(&Some(policy), syscall.as_ref()).is_ok());
+
+        let got_args = syscall
+            .as_any()
+            .downcast_ref::<TestHelperSyscall>()
+            .unwrap()
+            .get_mempolicy_args();
+
+        assert_eq!(got_args.len(), 1);
+        assert_eq!(got_args[0].mode, 2); // MPOL_BIND (corrected value)
+        assert_eq!(got_args[0].nodemask.len(), 1); // One c_ulong needed
+        assert_eq!(got_args[0].nodemask[0], 3); // 2^0 + 2^1 = 1 + 2 = 3
+        assert_eq!(got_args[0].maxnode, 2); // highest node ID (1) + 1 = 2
```

Comment:
Since `maxnode` is calculated as `(num_u64s * u64_bits) as u64` above, the test may be incorrect.